PA_implementation_overview.md

```diff
@@ -0,0 +1,66 @@
+
+
+## Phase 2 Bidding and Scoring Phase
+
+### Starting Worklet Executors
+Once all interest groups are loaded, Chrome starts loading the seller worklet, and all bidder worklets participating in an auction.  To load a worklet, Chrome first checks if there is already a matching worklet “executor” (Same worklet origin, same javascript URL, wasm URL, trusted signals URL, same frame - frame match is required by the DevTools hooks. "Frame" here means the frame runAdAuction() was called in.  It may be for a different auction / component auction, however).  If so, the executor is reused.  Note that different executors do not currently share any network fetches or Javascript contexts, although they do share the same global HTTP cache, so it’s generally a good idea to try and use the same URLs for all interest groups, to maximize executor reuse.  Also, interest groups that share an executor are guaranteed to be run in the priority order they were sorted into (this may change if trusted signals fetches are split up, though).  Interest groups that don’t share an executor are run on the basis of whichever has everything it needs to run first, runs first.
+
+Otherwise, Chrome requests a buyer or seller worklet process for the associated buyer/seller origin.  Buyers and sellers do not share processes, even if the origin is the same, though a single buyer or seller origin will globally use a single process shared by all of its executors. There are separate Chrome-wide limits on the number of buyer processes (10) and seller processes (3).  These numbers are not optimized, and subject to change.  On mobile, Chrome will generally reuse the single renderer process with one thread for bidders and one for sellers, instead of using more secure, isolated processes, due to resource constraints.  Within a process, all Protected Audience worklets use a single Javascript thread.  Once a process is assigned, Chrome creates an executor for the worklet, which starts loading the javascript and wasm URLs.  Because bidders share processes based on origin, this process means that all of a single buyer origin’s interest groups are assigned an executor (newly created or reused) at once.
+
+Bidders request processes/executors are requested in the order they appear in interestGroupBuyers, and new processes are assigned in a FIFO order.  So unless there’s a pre-existing bidder process for a particular origin, bids are generally generated based on the order buyers appear in the interestGroupBuyers.  There may be some potential advantages/disadvantages based on order (e.g., if Promises take a while to resolve, the cumulative timeout starts after executor assignment for bidders that get assigned an executor before the Promises resolve), so it’s recommended that buyer order be randomized.
```

Comment:
I accepted this, but I'm skeptical about adding an s after a backtick-ed phrase.  No matter how backticks are interpreted, it will come out looking weird.